docker-compose.yaml

```diff
@@ -0,0 +1,46 @@
+version: "3"
+services:
+ client:
+    container_name: journal
+    build: ./client
+    image: client
+    ports: 
+      - 8080:8080
+    working_dir: /code
+    volumes:
+      - type: bind
+        source: ./client
+        target: /code
+    command: [ "http-server" ]
+  api:
+    image: node:15.12
+    working_dir: /api
+    ports:
+      - "3000:3000"
+    environment:
+      - PGUSER=journaluser
+      - PGHOST=db
+      - PGPASSWORD=journalpass
+      - PGDATABASE=journaldb
+      - PGPORT=5432
+    volumes:
+      - type: bind
+        source: ./api
+        target: /api
+    command: bash -c "npm install && npm run seedDev && npm start"
+    depends_on:
+      - db
+
+  db:
+    image: postgres:12.4
+    ports:
+      - 5432:5432
+    environment:
+      - POSTGRES_USER=journaluser
+      - POSTGRES_DB=journaldb
+      - POSTGRES_PASSWORD=journalpass
+    volumes:
+      - "./db/migrations:/docker-entrypoint-initdb.d/:ro"
+
+volumes:
```

Comment:
is this redundant or does this do things because it is an empty volume with an empty dbdata